Jellyfin.Api/Controllers/SystemController.cs

```diff
@@ -42,20 +46,71 @@ public class SystemController : BaseJellyfinApiController
     /// <param name="fileSystem">Instance of <see cref="IFileSystem"/> interface.</param>
     /// <param name="networkManager">Instance of <see cref="INetworkManager"/> interface.</param>
     /// <param name="systemManager">Instance of <see cref="ISystemManager"/> interface.</param>
+    /// <param name="backupService">Instance of the <see cref="IBackupService"/> interface.</param>
     public SystemController(
         ILogger<SystemController> logger,
         IServerApplicationHost appHost,
         IServerApplicationPaths appPaths,
         IFileSystem fileSystem,
         INetworkManager networkManager,
-        ISystemManager systemManager)
+        ISystemManager systemManager,
+        IBackupService backupService)
     {
         _logger = logger;
         _appHost = appHost;
         _appPaths = appPaths;
         _fileSystem = fileSystem;
         _networkManager = networkManager;
         _systemManager = systemManager;
+        _backupService = backupService;
+    }
+
+    /// <summary>
+    /// Creates a new Backup.
+    /// </summary>
+    /// <response code="200">Backup created.</response>
+    /// <response code="403">User does not have permission to retrieve information.</response>
+    /// <returns>OK.</returns>
+    [HttpPost("Backup/Create")]
+    [Authorize(Policy = Policies.RequiresElevation)]
+    [ProducesResponseType(StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status403Forbidden)]
+    public async Task<IActionResult> CreateBackup()
+    {
+        await _backupService.CreateBackupAsync().ConfigureAwait(false);
+        return Ok();
+    }
+
+    /// <summary>
+    /// Restores to a backup by restarting the server and applying the backup.
+    /// </summary>
+    /// <param name="archivePath">The local path to the archive to restore from.</param>
+    /// <response code="200">Backup restore started.</response>
+    /// <response code="403">User does not have permission to retrieve information.</response>
+    /// <returns>OK.</returns>
+    [HttpPost("Backup/Restore")]
+    [Authorize(Policy = Policies.RequiresElevation)]
+    [ProducesResponseType(StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status403Forbidden)]
+    public IActionResult StartRestoreBackup(string archivePath)
+    {
+        _backupService.ScheduleRestoreAndRestartServer(archivePath);
+        return Ok();
```

Comment:
Empty responses should be 204 NoContent, also is it possible to make this blocking so a client knows then the backup is restored?

Reply:
No a restore includes a system restart which then in turn applies the restore and then restarts again.